### PR TITLE
Add further detection date filter values

### LINKED ISSUE #1437: new detection date filter values
Maybe we could add further detection date filter values to the current `today` and `yesterday` values.
- week (this week or 7 days)
- month (this month)
- year (this year)

### PR DESCRIPTION
**Pull request by csordasmarton**
Add further Detection date filter values:
* This week
* This month
* This year

Closes #1437

### CODE REVIEW

File: www/scripts/codecheckerviewer/filter/DateFilter.js [hunk 1]
```diff
@@ -214,6 +215,25 @@ function (declare, Deferred, dom, DateTextBox, TimeTextBox, SelectFilter,
           this.setDayInterval(yesterday);
           this.selectedItem = value;
           break;
+        case 'this week':
+          var today = new Date();
+          thisWeek = today.getDate() - today.getDay();
+          var monday = new Date(today.setDate(thisWeek));
```
Comment (whisperity) [inline www/scripts/codecheckerviewer/filter/DateFilter.js]: Not every locale has the week starting on Monday.

I'd rather see "this week", "this month" to be replaced on the gui with let's say "May 11-17", "August" and such.

File: www/scripts/codecheckerviewer/filter/DateFilter.js [hunk 2]
```diff
@@ -214,6 +215,25 @@ function (declare, Deferred, dom, DateTextBox, TimeTextBox, SelectFilter,
           this.setDayInterval(yesterday);
           this.selectedItem = value;
           break;
+        case 'this week':
```
Comment (whisperity) [inline www/scripts/codecheckerviewer/filter/DateFilter.js]: Isn't there a way to make this switch an integer-based one as opposed to string equality?

File: www/scripts/codecheckerviewer/filter/DateFilter.js (hunk 1 above)
Comment (gyorb) [reply www/scripts/codecheckerviewer/filter/DateFilter.js]: @whisperity @csordasmarton how about `last 7 days` instead of week?

**Comment by csordasmarton** [comment]
@gyorb Done.

**Pull request merged**